### ISSUE TITLE
`no-member` error when accessing a sub class variable with `self` returned from a super class method

### ISSUE THREAD

**Issue by MycroftKang** [PyCQA/pylint#7991]
### Bug description

Pylint reports a `no-member` error message when accessing a sub class variable with `self` returned from a super class method.

```python
class Foo:
    def save(self):
        return self


class Bar(Foo):
    def __init__(self) -> None:
        super().__init__()

        self.value = 12

    def save(self):
        return super().save()


print(Bar().save().value)
```

### Configuration

```ini
[MESSAGES CONTROL]

disable=R,C,W
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
temp.py:16:6: E1101: Instance of 'Foo' has no 'value' member (no-member)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 2.15.9
astroid 2.12.13
Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)]
```

### OS / Environment

Windows 11
